internal/ignition/ignition.go

```diff
@@ -130,6 +130,45 @@ IMAGE=$(echo $1 | sed 's/:.*//')
 podman images | grep $IMAGE || podman rmi --force $1 || true
 `
 
+const okdBinariesOverlayTemplate = `#!/bin/env bash
+set -eux
+# Fetch an image with OKD rpms
+RPMS_IMAGE="%s"
+while ! podman pull --quiet "${RPMS_IMAGE}"
+do
+    echo "Pull failed. Retrying ${RPMS_IMAGE}..."
+done
+mnt=$(podman image mount "${RPMS_IMAGE}")
+# Extract machine-config-daemon binary
+cp -rvf ${mnt}/binaries/machine-config-daemon /usr/local/bin/machine-config-daemon
+chmod a+x /usr/local/bin/machine-config-daemon
+restorecon -Rv /usr/local/bin/machine-config-daemon
+# Install RPMs in overlayed FS
+mkdir /tmp/rpms
+cp -rvf ${mnt}/rpms/* /tmp/rpms
+tmpd=$(mktemp -d)
+mkdir ${tmpd}/{upper,work}
+mount -t overlay -o lowerdir=/usr,upperdir=${tmpd}/upper,workdir=${tmpd}/work overlay /usr
+rpm -Uvh /tmp/rpms/*
+touch /tmp/.okd-rpms-installed
+podman rmi -f "${RPMS_IMAGE}"
+# Expand /var to 6G (3.2G on 16GB VM in insufficient)
+/bin/truncate -s 6G /run/ephemeral.xfsloop
+losetup -c /dev/loop0
+xfs_growfs /var
+mount -o remount,size=6G /run
+# Start agent
+systemctl start agent
```

Comment:
Isn't the agent already waiting on `/tmp/.okd-rpms-installed`? Why does it need a `start`?

Reply:
`agent` is not set to autorestart on failure, so it needs to be kickstarted. I suppose we could update the service, but not sure about unintended effects of this

Reply:
>autorestart on failure
What failure is it hitting? (assuming it's just waiting on your file to show up I wouldn't expect it to fail)

Reply:
>I suppose we could update the service, but not sure about unintended effects of this

That sounds like a good idea, let me verify if this wasn't done on purpose for some good reason

Reply:
Looks (f6403fb153d45bcb3204b61e28c410a6ec029fe2) like it was restarting early on, trying to find the commit where someone removed it

Reply:
>agent is not set to autorestart on failure

Oh wait this doesn't seem true
It is set to auto restart on anything, not just failure https://github.com/openshift/assisted-service/blob/70cd21e2b64f0e28993182c29cae733242e359a9/internal/ignition/ignition.go#L147

Reply:
It would be blocked by absence of `/tmp/.okd-rpms-installed` (which systemd treats as not started), but it won't autorestart

Reply:
I'm confused, if it's blocked by the absence of okd-rpms-installed, shouldn't systemd be watching for the existence of this file and start the service when that file shows up?

Or is that not how it works?

Reply:
>shouldn't systemd be watching for the existence of this file and start the service when that file shows up, automatically?

Not if you have Restart unset or disabled :/

Reply:
But it is set 
![image](https://user-images.githubusercontent.com/10882062/152786584-11390311-104c-46b8-b8a5-6c5b6f4c5e10.png)

Reply:
Ah, interesting. I'll play with that some more then

Reply:
Reworked in b84acd808584a68d6478b69121bbe59dd76eaf70 - OKD installs would add `Wants/After` dropin for agent service